### ISSUE TITLE
[30] Masked Autoencoders Are Scalable Vision Learners (MAE)

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#58]
[paper](https://arxiv.org/abs/2111.06377)
[PR12 진원님 리뷰](https://www.youtube.com/watch?v=mtUa3AAxPNQ)

Kaiming He 의 새로운 논문....!

# Intro
masked autoencoding 이 NLP와 VISION 에서 다른 이유가 뭘까? (BERT는 성공했는데, vision에서는 그닥..?)
이 질문에 대답하기 위해 3가지 관점으로 살펴본다.
1. vision은 CNN 이 정답이었는데, 요즘 ViT의 도입으로 architecture 들의 경계가 허물어짐
2. language 는 굉장히 semantic 하고 information-dense 하다. BERT 에서처럼 문장에서 몇 개만 mask 해서 맞추게 해도 충분하다. 근데 이미지는 그렇지 않다. 그냥 natural signal 이다. masked patch 예측은 물체에 대한 깊은 이해가 없어도 수행할 수 있는 꽤나 쉬운 task 이다.
3. autoencoder 의 decoder 는 단순 pixel 을 재구성하는 task 를 수행한다. low semantic 정보를 뽑아내는 거다. NLP 와는 반대됨. vision task 에서 decoder design은 semantic level 을 결정하는 중요한 요소이다.

# MAE
![image](https://user-images.githubusercontent.com/16400591/141944005-6e8498de-8826-496c-a8ad-cc377aba2eba.png)

진짜 simple 하다.
1. 75% 의 patch를 mask 한다.
2. encoder 에 mask 되지 않은 녀석들만 모아서 통과시킨다.
3. decoder 에는 mask 와 함께 (2) 의 output 을 함께 통과시켜서 이미지를 복원시킨다.
4. downstream을 풀어낼 때는 encoder 만 사용한다.

```
갓이밍 갓갓은 바로 SOTA를 찍어버렸다...!
```

decoder 는 light 하게 만들어서 pretraining 을 더 빠르게 만들었고, 
loss 는 MSE를 사용했다.

# Results
실험은 전체적으로 finetuning 하는 실험, 끝에 layer 하나 붙이는 linear probing 두 가지를 갖고 실험한다.

## Masking Ratio
![image](https://user-images.githubusercontent.com/16400591/142520530-b9babed1-7527-45ef-8dc8-eea6055ceb46.png)
imagenet 1K validation set 결과.
masking은 75% 정도가 적당했다.
![image](https://user-images.githubusercontent.com/16400591/142520936-b3fa9a84-842c-4976-b6f5-b3124712b28a.png)

## ViT-L vs MAE
최대한 fair comparison을 위해서, 비슷한 실험 setting으로 ViT Large 의 성능을 뽑았다.
강한 augmentation 이 ViT 에도 좋은 영향을 준다. #46 과도 일맥상통하는 내용.
![image](https://user-images.githubusercontent.com/16400591/142520774-cbe410fa-27a9-4a6a-b08d-5bfedb39690c.png)

## Ablation
여러가가지 ablation study 들을 수행했다.
![image](https://user-images.githubusercontent.com/16400591/142521317-e0324a51-4c5d-4f95-bdc0-fe87bac120ae.png)

`회색` 세팅이 default setting 들이다.
(a) decoder depth는 놀랍게도 1만 수행해도 finetuning 에서 굉장히 좋은 성능이 나온다.
(b) finetune 기준 128이면 충분
(c) encoder 통과시킬 때 mask 같이 넣으면 성능 떨어짐
(d) normalized pixel 을 reconstruction 시키는 게 가장 좋음
(e) crop, rand-size 로 좋은 성능 달성.
(f) masking 방법도 다양하게 수행해 보았는데, random이 제일 좋더라.
![image](https://user-images.githubusercontent.com/16400591/142529928-ff184aa2-a16b-4f08-b22a-407638eba6dd.png)

## Partial Fine-tuning
위 ablation 항목들을 보면 finetuning 결과와 linear probing 결과들이 경향이 일치하지 않는다.
물론 linear probing이 좋은 지표중 하나이지만, non-linear feature들을 평가하지 못한다.
이에 따라 partial fine-tuning 이라는 metric을 제안.
단순히 last-layer 에서부터 개수를 정해서 unfreeze 시키고 finetuning 시키는 방법이다.
![image](https://user-images.githubusercontent.com/16400591/142525886-501e8226-7bf2-4ca0-ae66-b774b7bb45ea.png)

그래프를 쭉 그려보면 backbone 비교가 용이하다.

## 과거 모델들과의 비교
![image](https://user-images.githubusercontent.com/16400591/141943570-bd536f44-e8bd-486c-94c5-66da361a08d7.png)
BEiT 보다 training 속도 3.5배 빠르고 dVAE 미리 학습시킬 필요 없다.

## 기타 downstream 수치들
![image](https://user-images.githubusercontent.com/16400591/142162096-04a6ee28-dd0e-476a-a089-e47c116faaed.png)
------------------------
![image](https://user-images.githubusercontent.com/16400591/142162214-485ac60a-ef01-40c4-9bf6-590d403ae737.png)
------------------------
![image](https://user-images.githubusercontent.com/16400591/142162270-77d7bcd6-ea0e-4ce6-8357-a669705de76f.png)

~아쉬운 점은 imagenet 1K 실험까지만 있다는 것.~

## Appendix
Experiment 세부 setting.
![image](https://user-images.githubusercontent.com/16400591/142161309-f9e8ee4c-b2ab-49a3-a4cf-100f7d47f8a5.png)
![image](https://user-images.githubusercontent.com/16400591/142161572-84944f94-c588-45d7-9b8c-46631fb3b849.png)
